src/lib/sqle/connection.service.ts

```diff
@@ -1,74 +1,223 @@
+  }
+  public get connections(): ISQLEConnection[] {
+    return this._connections;
+  }
+
+  public addConnectionAndConnect(connection: ISQLEConnection, options?: IConnectOptions): Observable<ISQLEConnection> {
+    if (this._getConnectionIndex(connection) > -1) {
+      throw Error('Connection already exists');
+    } else {
+      return this._pingAndSave(connection, true, options);
+    }
+  }
+
+  public addConnection(connection: ISQLEConnection, options?: IConnectOptions): Observable<ISQLEConnection> {
+    if (this._getConnectionIndex(connection) > -1) {
+      throw Error('Connection already exists');
+    } else {
+      return this._pingAndSave(connection, false, options);
+    }
+  }
 
-  public get current(): ISQLEConnection {
-    return current();
+  public connectToExistingConnection(
+    connection: ISQLEConnection,
+    options?: IConnectOptions,
+  ): Observable<ISQLEConnection> {
+    if (this._getConnectionIndex(connection) > -1) {
+      return this._pingAndSave(connection, true, options);
+    } else {
+      throw Error('Connection does not exist');
+    }
+  }
+  public removeConnection(connection: ISQLEConnection): ISQLEConnection {
+    const index: number = this._getConnectionIndex(connection);
+    if (index > -1) {
+      this._connections = [...this._connections.slice(0, index), ...this._connections.slice(index + 1)];
+      this._currentConnection =
+        this._currentConnection && this._connectionsAreEqual(this._currentConnection, connection)
+          ? undefined
+          : this._currentConnection;
+      return connection;
+    } else {
+      // connection does not exist but that is fine?
+      return undefined;
+    }
+  }
+
+  public disconnectFromCurrentConnection(): void {
+    this._currentConnection = undefined;
+  }
+
+  public removeAllConnections(): void {
+    this._connections = [];
+    this._currentConnection = undefined;
+  }
+
+  public connectionExits(connection: ISQLEConnection): boolean {
+    return this._getConnectionIndex(connection) > -1;
+  }
+
+  public connectionIsCurrentConnection(connection: ISQLEConnection): boolean {
+    return this._connectionsAreEqual(connection, this.currentConnection);
   }
 
-  public disconnect(): void {
-    sessionStorage.removeItem(CONNECTION_SESSION_KEY);
+  public stringifyConnection(connection: ISQLEConnection): string {
+    return stringifyConnection(connection);
   }
 
-  public connect(
+  private _pingAndSave(
     connection: ISQLEConnection,
-    opts?: { timeout: number; attempts: number },
+    setAsCurrent: boolean,
+    opts: IConnectOptions = { attempts: 2, timeout: 7000 },
   ): Observable<ISQLEConnection> {
-    // clear connection before starting a new one
-    this.disconnect();
     // test connection with SELECT 1
     return this._queryService.querySystem(connection, { query: 'SELECT 1;' }).pipe(
       // timeout connection if more than 7 seconds
-      timeout(opts?.timeout || 7000),
+      timeout(opts.timeout),
```

Comment:
Let's not remove the default just in case.

Reply:
The default is set in the param

Reply:
Actually reverted.